src/Data/Integer/Properties.agda

```diff
@@ -1735,28 +2079,242 @@ m≤m⊔n (+ m)    (+ n)    = +≤+ (ℕₚ.m≤m⊔n m n)
 n≤m⊔n : ∀ m n → n ≤ m ⊔ n
 n≤m⊔n m n rewrite ⊔-comm m n = m≤m⊔n n m
 
+mono-≤-distrib-⊔ : ∀ f → f Preserves _≤_ ⟶ _≤_ → ∀ m n → f (m ⊔ n) ≡ f m ⊔ f n
+mono-≤-distrib-⊔ f f-mono-≤ m n with <-cmp m n
+... | tri< m<n _    _   = trans (cong f (m≤n⇒m⊔n≡n (<⇒≤ m<n))) (sym (m≤n⇒m⊔n≡n (f-mono-≤ (<⇒≤ m<n))))
```

Comment:
This `trans (cong f X) (sym Y)` pattern occurs a lot... I'll ponder on how we might reduce the redundancy.

For starters I think you could probably shorten the proofs by a line by pattern matching on `≤-total m n` for the proofs where you're working with `_≤_` rather than `_<_`